### PR TITLE
OOP school library: unit tests 

### PR DESCRIPTION
**Pull request by uno36**
### Project requirements
- Add unit tests for all the methods in all the classes:
  - `Book`
  - `Person`
  - `Student`
  - `Teacher`
  - `Classroom`
  - `Rental`
  - all classes that you created while implementing the Decorator design pattern

### CODE REVIEW

File: spec/person_spec.rb
Comment (cynthiainga) [inline spec/person_spec.rb:1]: - [ ] Good job so far but there are still a few tests to write and you're good to go 😁 
  - You should test the scenario when the name of the person is not given to be equal to `Unknown` as it's the default value for the optional name parameter,
  - You should test the context when adding a new rental,
  - You should test the context when validating the name, for example, if the name is `alongnameiscorrected` it should be equal to `Alongnamei`.

File: spec/classroom_spec.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+require_relative '../classroom'
+
+RSpec.describe Classroom do
+  let(:classroom) { Classroom.new('101') }
+  let(:student) { Student.new(classroom: classroom, age: 19, name: 'John') }
+  # let(:classroom_student) { Classroom.new('101').add_student(student) }
```
Comment (cynthiainga) [inline spec/classroom_spec.rb]: - [ ] Please kindly delete the commented line of code to keep your code clean 🙃 
- [ ] Also as mentioned in the project requirements you should test all classes that you created while implementing the Decorator design pattern which means the `Trimmer, Capitalized, Nameable, and Decorator` should be tested as well

**Review by cynthiainga — CHANGES_REQUESTED** [review]
Hi @uno36 & @lorenzoworx & @akezeth, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- [ ] All test files are placed in a different directory `/spec` ✔️ 
- [ ] There is one file per class ✔️ 
- [ ] All implemented tests are passing ✔️ 

## Required Changes ♻️

- [ ] Please kindly check the comments under the review 👇🏾 👇🏾 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by OLIPLICHE — APPROVED** [review]
Hi @uno36, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Highlights
- Add unit tests to specific classes (✅)) 👏
- There is an 'it' block for each class method 👏👏 
- Work in pair programming 👏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**